lib/PasswordPolicyConfig.php

```diff
@@ -34,91 +42,109 @@ public function __construct(
 	 * get the enforced minimum length of passwords
 	 * @return non-negative-int
 	 */
-	public function getMinLength(): int {
-		return max(0, $this->appConfig->getValueInt(Application::APP_ID, 'minLength', 10));
+	public function getMinLength(?PasswordContext $context = null): int {
+		$key = $this->getScopedAppConfig('minLength', $context);
```

Comment:
Do I get this right that there is now only the scoped policies and no longer one default policy? If we do that we should add a migration step for existing configs. Otherwise we could have a default policy and fall back to that if no custom one is configured

Reply:
That is what is done, if e.g. there is no policy for sharing the default one is applied.

Reply:
Sorry then I misread parts of the code. Having another look.